src/type.js

```diff
@@ -192,12 +195,14 @@ async function typeImpl(
       prevWasMinus = false,
       prevWasPeriod = false,
       prevValue = '',
+      typedValue = '',
       eventOverrides,
     },
   ) {
     const key = char // TODO: check if this also valid for characters with diacritic markers e.g. úé etc
     const keyCode = char.charCodeAt(0)
     let nextPrevWasMinus, nextPrevWasPeriod
+    const textToBeTyped = typedValue + char
```

Comment:
We need to keep some var around to concat the chars since the `date` input only accepts valid dates for its value.